tox.ini

```diff
@@ -1,13 +1,22 @@
 [tox]
-envlist = py37,py38,pypy3,pre-commit
+# Platform specification support is available since version 2.0, see:
+# https://tox.wiki/en/latest/example/platform.html#basic-multi-platform-example
+minversion = 2.0
+envlist = {py37,py38,pypy3}-{unix,windows},pre-commit
 
 [testenv]
 deps = -rrequirements-dev.txt
 passenv = APPDATA HOME LOCALAPPDATA PROGRAMFILES RUSTUP_HOME
+platform =
+    unix: darwin|cygwin|linux\d*|aix\d*|sunos\d*|freebsd\d*
+    windows: win32
 commands =
-    coverage erase
-    coverage run -m pytest {posargs:tests}
-    coverage report
+    unix: coverage erase
+          coverage run -m pytest --basetemp="/tmp/pytest-{envname}" {posargs:tests}
+          coverage report
+    windows: coverage erase
+             coverage run -m pytest --basetemp="C:\tmp\pytest-{envname}" {posargs:tests}
+             coverage report
 
 [testenv:pre-commit]
 skip_install = true
```

Comment:
I'd rather not change this -- it's intentionally simple.  I meant to just add the temp override in `azure-pipelines.yml` since that's really the only place the paths are going to be so deep

Reply:
@asottile Thanks for taking care of this!